python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -504,6 +504,8 @@ def test_compiled_dag_ref_del(ray_start_regular):
         ref = compiled_dag.execute(1)
         del ref
 
+    compiled_dag.teardown()
```

Comment:
If we don't call `compiled_dag.teardown()`, the test suite will always have a segmentation fault.

![Uploading Screenshot 2024-06-21 at 4.36.16 PM.png…]()